src/automaton/examples/node/blockchain_cpp_node/blockchain_cpp_node_test.cc

```diff
@@ -44,8 +54,32 @@ int main() {
   std::shared_ptr<automaton::core::network::simulation> sim = automaton::core::network::simulation::get_simulator();
   sim->simulation_start(50);
   automaton::core::network::tcp_init();
-  testnet::create_testnet("blockchain", "testnet", "doesntmatter", testnet::network_protocol_type::simulation, 10,
-      create_connections_vector(10, 2));
+  testnet::create_testnet("blockchain", "testnet", "doesntmatter", testnet::network_protocol_type::simulation, 1000,
+      create_connections_vector(1000, 4));
+
+  std::vector<std::thread> worker_threads;
+  for (uint32_t i = 0; i < WORKER_NUMBER; ++i) {
+    worker_threads.push_back(std::thread([&]() {
+      while (worker_running) {
+        std::vector<std::string> node_ids = node::list_nodes();
+        for (uint32_t i = 0; i < node_ids.size(); ++i) {
+          if (!worker_running) {
+            break;
+          }
+          auto node = node::get_node(node_ids[i]);
```

Comment:
There is nothing here to prevent two worker threads from initiating a process_update on the same node.

Reply:
`  
if (current_time >= (node->get_time_to_update())) {
              node->process_update(current_time);
            }
`

Reply:
But if two threads get to this if at the same time, they'll both proceed to do ``process_update``